packages/backend/server/src/core/workspaces/resolvers/workspace.ts

```diff
@@ -336,6 +338,11 @@ export class WorkspaceResolver {
       throw new ForbiddenException('Cannot change owner');
     }
 
+    const lockFlag = `invite:${workspaceId}`;
+    if (!(await this.mutex.lock(lockFlag))) {
```

Comment:
make the execution after lock acquired as a callback? there are sometimes lots of return paths. I concerns there might be chances to forget to release the locks.